extensions/default/src/DicomWebDataSource/index.js

```diff
@@ -438,19 +438,37 @@ function createDicomWebApi(dicomWebConfig, servicesManager) {
             instance,
           });
 
+          const isMultiframe = instance.NumberOfFrames > 1;
```

Comment:
Can you extract a common parent class rather than repeating the code?  That would help new implementers with adding new data source types.